python/pyspark/sql/worker/plan_data_source_read.py

```diff
@@ -51,6 +52,71 @@
 )
 
 
+def records_to_arrow_batches(
+    output_iter: Iterator[Tuple],
+    max_arrow_batch_size: int,
+    return_type: StructType,
+    data_source: DataSource,
+) -> Iterable[pa.RecordBatch]:
```

Comment:
Let's add some docstring for this function.

Reply:
docstring added.